src/Microsoft.AspNet.Mvc/MvcServices.cs

```diff
@@ -151,8 +151,10 @@ public static IEnumerable<IServiceDescriptor> GetDefaultServices(IConfiguration
             // IActionBindingContextProvider. Therefore it too is scoped.
             yield return describe.Transient<IHtmlGenerator, DefaultHtmlGenerator>();
 
-            yield return describe.Transient<IViewComponentSelector, DefaultViewComponentSelector>();
+            // These do caching so they should stay singleton
```

Comment:
nit: could / should the caching for view components be moved into a single singleton?